htdocs/lockaccount.php

```diff
@@ -24,7 +24,46 @@
     $ldap = $ldap_connection[0];
     $result = $ldap_connection[1];
 
-    if ($ldap) {
+
+    # Consider pwdLockout = false by default
+    $pwdLockout = "false";
+
+    # Search pwdLockout in associated ppolicy
+    if ($ldap)
+    {
+        $search_ppolicysubentry = ldap_read($ldap, $dn, "(objectClass=*)", array('pwdpolicysubentry'));
+        $user_entry = ldap_get_entries($ldap, $search_ppolicysubentry);
+
+
+        # Search active password policy
+        $pwdPolicy = "";
+        if (isset($user_entry[0]['pwdpolicysubentry'][0])) {
+            $pwdPolicy = $user_entry[0]['pwdpolicysubentry'][0];
+        } elseif (isset($ldap_default_ppolicy)) {
+            $pwdPolicy = $ldap_default_ppolicy;
+        }
+
+        $ppolicy_entry = "";
+
+        # Search pwdLockout attribute
+        if ($pwdPolicy) {
+                $search_ppolicy = ldap_read($ldap, $pwdPolicy, "(objectClass=pwdPolicy)", array('pwdlockout'));
+
+                if ( $errno ) {
+                    error_log("LDAP - PPolicy search error $errno  (".ldap_error($ldap).")");
+                } else {
+                    $ppolicy_entry = ldap_get_entries($ldap, $search_ppolicy);
+                    $pwdLockout = strtolower($ppolicy_entry[0]['pwdlockout'][0]);
+                    if(! isset($pwdLockout) or $pwdLockout == "" or $pwdLockout == "false" )
```

Comment:
You should just check if pwdLockout is false after it has been converted to a PHP boolean